src/main/java/spoon/reflect/visitor/LexicalScopeBuilder.java

```diff
@@ -21,21 +24,21 @@
 import spoon.reflect.declaration.CtNamedElement;
 
 /**
- * The {@link EarlyTerminatingScanner} implementation,
- * which knows mapping of simple names to elements in the actually scanned scope.
+ * Responsible for building lexical scopes.
  *
- * @param <T> the type of the result produced by this scanner.
  */
-public class NameScopeScanner<T> extends EarlyTerminatingScanner<T> {
-	private final Deque<NameScope> scopes = new ArrayDeque<>();
+public class LexicalScopeBuilder extends EarlyTerminatingScanner<Object> {
+	private final List<LexicalScope> allScopes = new ArrayList<>();
```

Comment:
I do not think that collecting of all scopes makes sense.
Note that scopes are even changing during scanning. For example:
```java
void draw() {
 //scope1
int a;
//scope2
int b;
//scope3
}
```
while scanning the statements of method `draw`, there is only one LexicalScope object whose content is different while scanning each comment.

In all my current use cases (handling of imports #2683) I am scanning AST and I need current lexical scope only.

If you see an use case behind I can live with that. I hope that amount of collected elements is not too big to consume much memory... :-)

Reply:
Thanks for the explanation. Fixed.